src/Symfony/Component/Mime/Header/Headers.php

```diff
@@ -122,6 +122,22 @@ public function addParameterizedHeader(string $name, string $value, array $param
         return $this->add(new ParameterizedHeader($name, $value, $params));
     }
 
+    /**
+     * @return $this
+     */
+    public function addTag(string $value): self
+    {
+        return $this->add(new TagHeader($value));
+    }
+
+    /**
+     * @return $this
+     */
+    public function addMetadata(string $key, string $value): self
+    {
+        return $this->add(new MetadataHeader($key, $value));
+    }
+
```

Comment:
I would add the header classes in the Mailer component and not add anything in the Mime component (as these headers are non-standard ones).